library/std/src/fs.rs

```diff
@@ -116,6 +116,13 @@ pub struct File {
     inner: fs_imp::File,
 }
 
+#[unstable(feature = "dirfd", issue = "120426")]
+#[cfg(target_family = "unix")]
+/// An object providing access to a directory on the filesystem.
```

Comment:
Could you elaborate the documentation here a bit? At least make it clear that this is a handle, and what happens on drop (similar to `File`).